src/llvmir2hll/graphs/cfg/cfg_traversal.cpp

```diff
@@ -163,12 +176,78 @@ bool CFGTraversal::getCurrRetVal() const {
 * @param[in] node Node to be traversed.
 * @param[in] stmtIter Iterator to a statement in @a node to be checked.
 *
-* If @a stmtIter equals @c node->stmt_end(), the function traverses all
-* successors of @a node.
+* This function traverses the entire CFG rooted at @a node, starting with the statement
+* in @a stmtIter.
 */
 bool CFGTraversal::performTraversalImpl(ShPtr<CFG::Node> node,
 		CFG::stmt_iterator stmtIter) {
-	while (stmtIter != node->stmt_end()) {
+	// The stack nodes consist of the current CFG node, a statement iterator storing the current
+	// place in the node's statements, and an iterator storing the current place in the node's
+	// successor list.
+	// Statement place is only needed because we may be called with a statement iterator
+	// that doesn't point to the beginning.
+	using NodeType = std::tuple<ShPtr<CFG::Node>, CFG::stmt_iterator, CFG::succ_iterator>;
+	std::stack<NodeType> visitStack;
+	visitStack.push({node, stmtIter, node->succ_begin()});
```

Comment:
The Linux TravisCI build [has failed](https://travis-ci.org/avast-tl/retdec/jobs/490297897#L2276) on this line:
```
src/llvmir2hll/graphs/cfg/cfg_traversal.cpp: In member function ‘bool retdec::llvmir2hll::CFGTraversal::performTraversalImpl(retdec::llvmir2hll::ShPtr<retdec::llvmir2hll::CFG::Node>, retdec::llvmir2hll::CFG::stmt_iterator)’:
src/llvmir2hll/graphs/cfg/cfg_traversal.cpp:191:54: error: converting to ‘std::stack<...>::value_type’ from initializer list would use explicit constructor ‘constexpr std::tuple<>::tuple(...) [...]’
```
Could you please verify? It uses GCC 4.9, which is the oldest GCC version that we support. If you are unable to reproduce this, let me know and I will take a look.

Reply:
I was able to reproduce this fail in a debian-jessie Docker container. Switching from `push()` to `emplace()` and removing the curly brackets (see my other comments) resolves the compilation issue on GCC 4.9, so this is what I suggest doing.

Reply:
The TravisCI build runs just fine, so I am marking this as resolved.